zqd/zeek/zeek.go

```diff
@@ -0,0 +1,91 @@
+package zeek
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// ExecScript will be fed into a launched zeek process as the --exec option. The
+// default script disables the packet_filter and loaded scripts logs. These logs
+// are disabled because the emit either timeless logs or logs with timestamp
+// set to execution time rather than time of capture.
+var ExecScript = `
+event zeek_init() {
+	Log::disable_stream(PacketFilter::LOG);
+	Log::disable_stream(LoadedScripts::LOG);
+}`
+
+// ErrNotFound is returned from LauncherFromPath when the zeek executable is not
+// found.
+var ErrNotFound = errors.New("zeek not found")
+
+// Process is an interface for interacting running with a running zeek process.
+type Process interface {
+	// Wait waits for a running process to exit, returning any errors that
+	// occur.
+	Wait() error
+}
+
+// Launcher is a function when fed a context, pcap reader stream, and a zeek
+// log output dir, will return a running zeek process. If there is an error
+// starting the Process, that error will be returned.
+type Launcher func(context.Context, io.Reader, string) (Process, error)
```

Comment:
Nit: No need for the Process interface if you change `type process` to `type Process` and change Launcher to return `(*Process, error)`.

Reply:
It would not allow this to be testable then and would defeat the purpose of this pr.